consensus-types/types/block.go

```diff
@@ -63,10 +63,10 @@ func NewBeaconBlockWithVersion(
 	slot math.Slot,
 	proposerIndex math.ValidatorIndex,
 	parentBlockRoot common.Root,
-	forkVersion uint32,
+	forkVersion common.Version,
 ) (*BeaconBlock, error) {
 	switch forkVersion {
-	case version.Deneb, version.Deneb1:
+	case version.Deneb(), version.Deneb1():
```

Comment:
nit: not sure we should do this.
I think this works for use because:
- the underlying type of `common.Version` is  a byte slice
- we only use comparison not ordering

Still maybe we should introduce an explitic version.Equal helper?
It may be my C++ mind speaking here, this is probably fine

Reply:
.Equals (among other comparable funcs) implemented [here](https://github.com/berachain/beacon-kit/blob/5ec0a3ded8a773388704e7afd247b6c19f2228a9/primitives/bytes/b4.go#L131)